libjulia/backends/evm/EVMCodeTransform.cpp

```diff
@@ -217,6 +217,18 @@ void CodeTransform::operator()(assembly::Instruction const& _instruction)
 	checkStackHeight(&_instruction);
 }
 
+void CodeTransform::operator()(If const& _if)
+{
+	visitExpression(*_if.condition);
+	m_assembly.appendInstruction(solidity::Instruction::ISZERO);
+	AbstractAssembly::LabelID end = m_assembly.newLabelId();
+	m_assembly.appendJumpToIf(end);
+	(*this)(_if.body);
+	m_assembly.setSourceLocation(_if.location);
```

Comment:
Shouldn't the sourcelocation be before the body or even the condition?

Reply:
Hm, the source location of the label is correct, but what is wrong is the source location of the ISZERO and the jump above.

Reply:
Will you look into this?

Reply:
Done.